fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java

```diff
@@ -0,0 +1,453 @@
+            Set<String> indexNonAggregatedColumnNames = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+            entry.getValue().stream().filter(column -> !column.isAggregated())
+                    .forEach(column -> indexNonAggregatedColumnNames.add(column.getName()));
+            if (!indexNonAggregatedColumnNames.containsAll(columnsInPredicates)) {
+                iterator.remove();
+            }
+        }
+        LOG.debug("Those mv pass the test of compensating predicates:"
+                          + Joiner.on(",").join(candidateIndexIdToSchema.keySet()));
+    }
+
+    /**
+     * View      Query        result
+     * SPJ       SPJG OR SPJ  pass
+     * SPJG      SPJ          fail
+     * SPJG      SPJG         pass
+     * 1. grouping columns in query is subset of grouping columns in view
+     * 2. the empty grouping columns in query is subset of all of views
+     *
+     * @param columnsInGrouping
+     * @param candidateIndexIdToSchema
+     */
+
+    private void checkGrouping(Set<String> columnsInGrouping, Map<Long, List<Column>> candidateIndexIdToSchema) {
+        Iterator<Map.Entry<Long, List<Column>>> iterator = candidateIndexIdToSchema.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<Long, List<Column>> entry = iterator.next();
+            Set<String> indexNonAggregatedColumnNames = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+            List<Column> candidateIndexSchema = entry.getValue();
+            candidateIndexSchema.stream().filter(column -> !column.isAggregated())
+                    .forEach(column -> indexNonAggregatedColumnNames.add(column.getName()));
+            // When the candidate index is SPJ type, it passes the verification directly
+            if (indexNonAggregatedColumnNames.size() == candidateIndexSchema.size()) {
+                continue;
+            }
+            // When the query is SPJ type but the candidate index is SPJG type, it will not pass directly.
+            if (isSPJQuery || disableSPJGView) {
+                iterator.remove();
+                continue;
+            }
+            // The query is SPJG. The candidate index is SPJG too.
+            // The grouping columns in query is empty. For example: select sum(A) from T
+            if (columnsInGrouping == null) {
+                continue;
+            }
+            // The grouping columns in query must be subset of the grouping columns in view
+            if (!indexNonAggregatedColumnNames.containsAll(columnsInGrouping)) {
+                iterator.remove();
+            }
+        }
+        LOG.debug("Those mv pass the test of grouping:"
+                          + Joiner.on(",").join(candidateIndexIdToSchema.keySet()));
+    }
+
+    private void checkAggregationFunction(Set<AggregatedColumn> aggregatedColumnsInQueryOutput,
+                                          Map<Long, List<Column>> candidateIndexIdToSchema) {
```

Comment:
No, it can't.
If the aggregatedColumnInQueryOutput is null or empty, the query will be a SPJ.
But the candidate index such as SPJG should be filter in the following code.